### DISCUSSION TITLE
nodejs.org denies access to zip files

### DISCUSSION THREAD

**Discussion by askingtherightquestion** [nodejs/help#4708] [Q&A]
Hello everyone

Our git pipelines isn't able to access http://nodejs.org/dist/ since Saturday: 

```
> Could not resolve all files for configuration ':detachedConfiguration1'.
   > Could not resolve org.nodejs:node:16.13.2.
     Required by:
         project :
      > Could not resolve org.nodejs:node:16.13.2.
         > Could not get resource 'http://nodejs.org/dist/v16.13.2/ivy.xml'.
            > Could not GET 'http://nodejs.org/dist/v16.13.2/ivy.xml'. Received status code 403 from server: Forbidden
```

The issue isn't caused by the missing ivy.xml. This issue suggests that the error is also thrown if the script isn't able to fetch the zip file: https://github.com/srs/gradle-node-plugin/issues/292

This issue seems to impact other people / projects as well:
- https://github.com/nodejs/node/discussions/47124 (already closed)
- https://stackoverflow.com/questions/75795920/could-not-get-resource-https-nodejs-org-dist-v14-19-3-ivy-xml

One answer suggests that it might have something to do with DDNS prevention software installed by http://nodejs.org/. Does anyone face the same issue? Do you have any suggestions on how to solve this?

PS: The subfolders of http://nodejs.org/dist/ returned the HTTP error 500 a couple of hours ago. I tested this with different IPs and computers in order to make sure that I haven't been blocked or anything. Can I see somewhere if changes were done to the network of nodejs.org?

**Comment by targos** [comment]
I just checked in our Cloudflare logs and out of 1.14 million requests for zip files in the last 3 days, only 24 were blocked, all for the path `/nodejs.org.sql.zip` (and not from Gradle).

**Reply by askingtherightquestion** [reply]
Thank you very much for the fast response!

The requests would have probably been made directly from the GitLab servers. I don't have direct access to them, but that would be something like git.n....s.com in my case without giving too much away.

I am working actively on this issue in order to find the cause. The problem might as well be a proxy or something else. I just found it odd that other people reported the same issue around the same time.

**Reply by askingtherightquestion** [reply]
The problem seems to be related to https://github.com/nodejs/build/issues/3223 and https://github.com/nodejs/nodejs.org/issues/5149.

We're using a plugin called "gradle-node-plugin" in our project. I'm not familiar with the inner workings of the plugin, but it seems to have some connection to JFrog (https://github.com/srs/gradle-node-plugin/blob/master/gradle/publishing.gradle). The error message regarding the ivy-File might be misleading. I suspect that the plugin checks first for the ivy file, then for the zip file and finally returns an error message only mentioning the ivy file. A 404 is handled correctly, but a 403 seems to be unexpected... I will try to update the gradle-node-plugin tomorrow and check if it solves our problem.

I'm not a frontend developer and I'm not sure if we're even supposed to download anything from nodejs.org. The CDN should theoretically be able to handle most of the requests, but there seems to be a reason if you had to activate Cloudflare rules. I'm genuinely happy that this didn't happen a week earlier - this would have had a major impact on a bigger prod release. It just seems crazy to me how the whole ecosystem of node/npm is supposed to work, if we're dependent on external libraries.

**Reply by UltraL0rd** [reply]
Since Saturday I have been receiving the following when attempting to build via bitbucket pipeline

Execution failed for task ':nodeSetup'.
> Could not resolve all dependencies for configuration ':detachedConfiguration16'.
   > Could not resolve org.nodejs:node:17.0.1.
     Required by:
         project :
      > Could not resolve org.nodejs:node:17.0.1.
         > Could not get resource 'https://nodejs.org/dist/v17.0.1/ivy.xml'.
            > Could not GET 'https://nodejs.org/dist/v17.0.1/ivy.xml'. Received status code 403 from server: Forbidden

A GET call via POSTMAN produces the same results.

**Reply by UltraL0rd** [reply]
@askingtherightquestion did your update fix the problem? I'm stuck trying to build code that I need to get live

**Reply by askingtherightquestion** [reply]
I was able to resolve it by replacing the underlying node plugin we were using in gradle. Check if you have a build.gradle file in your project.

I had to replace `classpath 'com.moowork.gradle:gradle-node-plugin:1.1.1'` with `classpath 'com.github.node-gradle:gradle-node-plugin:1.5.3'`.

**Reply by UltraL0rd** [reply]
@askingtherightquestion already using classpath "com.github.node-gradle:gradle-node-plugin:2.2.4" still getting the ivy.xml error.

**Reply by UltraL0rd** [reply]
@targos Looks like my issue is slightly different from @askingtherightquestion . 2 days ago my pipelines on bitbucket and locally where still building fine. Yesterday evening my local and pipelines started producing the following errors :
  > Could not resolve org.nodejs:node:17.0.1.
     Required by:
         project :
      > Could not resolve org.nodejs:node:17.0.1.
         > Could not get resource 'https://nodejs.org/dist/v17.0.1/ivy.xml'.
            > Could not GET 'https://nodejs.org/dist/v17.0.1/ivy.xml'. Received status code 403 from server: Forbidden
            
 The version doesn't matter, they all throw the same error

**Reply by askingtherightquestion** [reply]
The maintainer of nodejs.org are referencing this comment as solution: https://github.com/srs/gradle-node-plugin/issues/292#issuecomment-562848790

I would just try out, if the version 2.2.0 works in a first step. The error happens within the plugin. It doesn't expect a 403 response.

The timing of your errors doesn't indicate that there were new changes. It worked maybe until yesterday, because of cached results (CDN, repository solution, local installation).
